File: dt-spark-core/src/main/scala/com/kevin/scala/actions/Count_RDD.scala
package com.kevin.scala.actions

import org.apache.spark.{SparkConf, SparkContext}

/**
  * 返回结果集中的行数
  */
object Count_RDD {

  def main(args: Array[String]):Unit = {

    val file = "DTSparkCore\\src\\main\\resources\\test.txt"
    // 1.创建SparkConf
    val conf = new SparkConf().setAppName("Count_RDD").setMaster("local")
    // 2.创建SparkContext
    val sc = new SparkContext(conf)
    // 3.统计数据记录数
    val count = sc.textFile(file).count()
    // 4.打印
    println(count)
    // 5.关闭sc
    sc.stop()

  }

}

File: dt-spark-core/src/main/java/com/kevin/java/actions/Count_RDD.java
package com.kevin.java.actions;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * @author kevin
 * @version 1.0
 * @description     返回结果集中的行数
 * @createDate 2018/12/27
 */
public class Count_RDD {

    public static void main(String[] args) {

        // 1.创建SparkConf，设置作业名称和模式
        SparkConf conf = new SparkConf().setAppName("Count_RDD").setMaster("local");

        // 2.JavaSparkContext是Spark的唯一通道
        JavaSparkContext sc = new JavaSparkContext(conf);

        String file = "DTSparkCore\\src\\main\\resources\\test.txt";

        // 4.获取文件数据
        JavaRDD<String> liens = sc.textFile(file);

        // 5.统计其行数
        long count = liens.count();

        System.out.println(count);

        // 6.关闭JavaSparkContext
        sc.close();
    }
}
